screendoor/parseapplication.py

```diff
@@ -128,16 +118,66 @@ def fill_in_single_line_arguments(item, applicant):
 
 
 def correct_split_item(idx, tables, item):
-    if ((idx + 1) != len(tables)) and not str(item.shape) == "(1, 1)":
+    if ((idx + 1) < len(tables)) and not str(item.shape) == "(1, 1)":
         item2 = tables[idx + 1]
 
         if item2.empty:
             return item
-
-        if item2.iloc[0, 0] == "NaN":
+        if item2.ix[0, 0] == "NaN":
             item.iloc[-1, -1] = item.iloc[-1, -1] + item2.iloc[0, 1]
+            tables[idx] = item
+
+
+            del tables[idx + 1]
+    return tables
+
+
+def check_if_table_valid(table):
```

Comment:
For code like this you can do:

`def check_if_table_valid(table):
    return not isinstance(table, pd.DataFrame) or table.empty`

and it should be the same